### PR TITLE
Adjust the Meta1Prefix, Meta2Prefix and SystemPrefix definitions.

### PR DESCRIPTION
**Pull request by petermattis**
Meta1Prefix  -> \x02
Meta2Prefix  -> \x03
SystemPrefix -> \x04

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3335)

### CODE REVIEW

**Comment by es-chow** [comment]
Could you please give some explanation on why make this change? Thanks.

**Comment by petermattis** [comment]
@es-chow It saves a very tiny bit of space per-key for the encoded system and meta keys. Is it problematic in some way?

File: keys/constants.go [hunk 1]
```diff
@@ -119,19 +119,16 @@ var (
...
-	SystemMax    = roachpb.Key("\x03")
-
-	// MetaPrefix is the prefix for range metadata keys. Notice that
-	// an extra null character in the prefix causes all range addressing
-	// records to sort before any system tables which they might describe.
-	MetaPrefix = MakeKey(SystemPrefix, roachpb.RKey("\x01m"))
-	// Meta1Prefix is the first level of key addressing. The value is a
-	// RangeDescriptor struct.
-	Meta1Prefix = roachpb.Key(MakeKey(MetaPrefix, roachpb.RKey("1")))
+	SystemPrefix = roachpb.Key("\x04")
```
Comment (bdarnell) [inline keys/constants.go]: Move this down below the meta definitions to keep the order clear.

File: keys/constants.go [hunk 2]
```diff
@@ -164,6 +163,9 @@ var (
 	// TableDataPrefix prefixes all table data. It is specifically chosen to
 	// occur after the range of common user data prefixes so that tests which use
 	// those prefixes will not see table data.
+	//
+	// TODO(peter): Replace with TableDataMin and TableDataMax which are the min
```
Comment (bdarnell) [inline keys/constants.go]: `encoding.intMin` now falls in the meta range.

**Comment by bdarnell** [comment]
I don't think the space savings alone are sufficient justification for this change - there simply aren't enough meta keys to matter. I like the cleanliness of making all the magic prefixes non-overlapping, but now that I see the comment about TableDataMin I think it will make more difference in the long run to allow table data to use raw int prefixes, so we should make sure that that is possible.

**Comment by es-chow** [comment]
@petermattis, No issue for me now. Just wonder if table data in SQL layer cannot use the raw tableID as prefix after this change?

**Comment by petermattis** [comment]
@es-chow We'll be able to use the raw tableID as the prefix after this change with a minor adjustment to the tags used in the encoding. I started working on that but ran into a a few test failures that I've been unable to fix yet. But be assured that is the plan and that this change does nothing to prohibit it.

File: keys/constants.go (hunk 2 above)
Comment (petermattis) [inline keys/constants.go]: Yeah. That isn't actually a problem for table IDs which use `EncodeUvarint`, but safest to adjust the definition of `intMin`. As mentioned elsewhere, I started working on this TODO and ran into variety of test failures which I haven't dug to the bottom of yet.

**Comment by es-chow** [comment]
@petermattis, Thanks for your clarification.

**Pull request merged**